pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -682,4 +692,75 @@ public List<ThreadSafeReportListener> getListeners() {
     public void addListeners(List<ThreadSafeReportListener> allListeners) {
         listeners.addAll(allListeners);
     }
+
+    /**
+     * Creates a new report taking all the information from this report,
+     * but filtering the violations.
+     *
+     * @param filter when true, the violation will be kept.
+     * @return copy of this report
+     */
+    public Report filterViolations(Predicate<RuleViolation> filter) {
```

Comment:
Maybe we should mark this `@Experimental` as we expect the `Predicate` to change to the standard jutil.function type in PMD 7?

Reply:
Good idea